src/realm/realmReducer.js

```diff
@@ -239,6 +249,15 @@ export default (
             if (data.description !== undefined) {
               result.description = data.description;
             }
+            if (data.mandatory_topics !== undefined) {
+              result.mandatoryTopics = data.mandatory_topics;
+            }
```

Comment:
I looked at the `RealmUpdateDictEvent` type to see what that event contains, and that led me to the `RealmDataForUpdate` type, which goes like this:

```js
  add_custom_emoji_policy:
    InitialDataRealm['realm_add_custom_emoji_policy'],
  allow_community_topic_editing: // deprecated
    InitialDataRealm['realm_allow_community_topic_editing'],
  allow_edit_history:
    InitialDataRealm['realm_allow_edit_history'],
  // etc.
```

I went down that list, and for each item that we were storing in the realm state, I checked if we were also handling the event. For example, for that third item, `realm_allow_edit_history` appears in `realmReducer`'s `REGISTER_COMPLETE` handler, and `allow_edit_history` appears in `realmReducer`'s realm-event op-update handler, so that one is taken care of.

None of `realm_non_active_users`, `realm_filters`, `realm_emoji`, `realm_default_external_accounts`, `realm_push_notifications_enabled`, and `server_web_public_streams_enabled` appear in `RealmUpdateDictEvent`, so we shouldn't expect them to be updated with this event.

- I don't know if we can get updates to `nonActiveUsers`.
- It looks like we get updates to `filters` with our handling of the `realm_filters` event (via our Redux action with type `EVENT_REALM_FILTERS`).
- It looks like we get updates to `emoji` with our handling of the `realm_emoji` event (via our Redux action with type `EVENT_REALM_EMOJI_UPDATE`).
- There's no event we'd look to for `defaultExternalAccounts`; see ee74a94812de5429d11d0a06351b572651e9b912 and [discussion](https://chat.zulip.org/#narrow/stream/378-api-design/topic/.60value.60.20in.20realm.20op.3A.20update/near/1276269).
- I suspect there's likewise no event for `pushNotificationsEnabled` but I'm not sure.
- I don't think there's an event for `webPublicStreamsEnabled`. The initial-data property is `server_web_public_streams_enabled`, and since it's prefixed with `server_` I'd imagine it can only be changed with a server restart, and we should handle restarts by reregistering (that's #4793).

Reply:
Very helpful, thanks!

> None of `realm_non_active_users`, […], and `server_web_public_streams_enabled` appear in `RealmUpdateDictEvent`, so we shouldn't expect them to be updated with this event.

Looking through just now, I think I find one more: `realm_enable_read_receipts`. (Unless it should appear in `RealmUpdateDictEvent` after all? Seems like the sort of thing I'd expect to be there.)


> * I don't know if we can get updates to `nonActiveUsers`.

I think the natural way for that to be updated is as a user-related event. Like `user/update`:
https://zulip.com/api/get-events#realm_user-update
That doesn't mention updates to `is_active`, though. I wonder if those do happen and it's just undocumented, or if there isn't an event for that happening (i.e., for a user getting deactivated or reactivated), or what. Hmm.



> * I suspect there's likewise no event for `pushNotificationsEnabled` but I'm not sure.

I agree. I think `realm_push_notifications_enabled` is really a server-level setting, which can only change on server restart.

---

Because the bulk of the RealmState properties -- or specifically, those that we've grouped together as coming from `InitialDataRealm` -- are found in `RealmDataForUpdate` and are handled in this spot, I think a good way to make this auditable would be to get them all listed here: when they aren't handled here, to have a line or two of comments saying why not, e.g. where they're handled instead.

So let's remember to do that as a followup.

Reply:
> Looking through just now, I think I find one more: `realm_enable_read_receipts`. (Unless it should appear in `RealmUpdateDictEvent` after all? Seems like the sort of thing I'd expect to be there.)

I do see it there:

```js
  enable_read_receipts:
    InitialDataRealm['realm_enable_read_receipts'],
```

and I do see handling in `realmReducer` for the initial data and for the event. Have I misunderstood something? 🙂

Reply:
Hmm, yeah.

That one was added relatively recently, in 03f7a98a92e84533a45202edc5e31fbf4b158124 a few weeks ago (in all the different places it appears.) Perhaps I was somehow looking at an out-of-date version of that file? I remember thinking of that hypothesis and specifically checking for it, though. :shrug: Anyway, that one looks good.